File: Practice_Set/app/src/main/java/com/example/practice_set/MainActivity.java
package com.example.practice_set;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.KeyEvent;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    TextView tvFirst,tvWay;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvFirst=findViewById(R.id.btnFirst);
        tvWay=findViewById(R.id.btnFirstWay);

   tvFirst.setOnClickListener(new View.OnClickListener() {
       @Override
       public void onClick(View v) {
           Toast.makeText(MainActivity.this, "First Layout", Toast.LENGTH_SHORT).show();
       }
   });
   tvWay.setOnClickListener(new View.OnClickListener() {
       @Override
       public void onClick(View v) {
           Toast.makeText(MainActivity.this, "First Way", Toast.LENGTH_SHORT).show();
       }
   });
    }
}

File: AddingInRecyclerView/app/src/main/java/com/example/addinginrecyclerview/MainActivity.kt
package com.example.addinginrecyclerview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.addinginrecyclerview.model.UserData
import com.example.addinginrecyclerview.view.UserAdaptar
import com.google.android.material.floatingactionbutton.FloatingActionButton

class MainActivity : AppCompatActivity() {
    private lateinit var addsBtn:FloatingActionButton
    private lateinit var recy:RecyclerView
    private lateinit var userList:ArrayList<UserData>
    private lateinit var userAdapater:UserAdaptar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // set List
        userList = ArrayList()
        // set findViewbyID
        addsBtn = findViewById(R.id.fbBtn)
        recy = findViewById(R.id.rCAdd)
        //  set Adapter
        userAdapater = UserAdaptar(this,userList)
        // set Recycler Adapter
        recy.layoutManager= LinearLayoutManager(this)
        recy.adapter = userAdapater
        // set DialogBox
        addsBtn.setOnClickListener{ addInfo() }

    }

    private fun addInfo() {
        val inflater = LayoutInflater.from(this)
        val v = inflater.inflate(R.layout.add_item,null)
        val addDialog = AlertDialog.Builder(this)
        // set View
        val userName = v.findViewById<EditText>(R.id.etName)
        val userNo = v.findViewById<EditText>(R.id.etNumber)
        addDialog.setView(v)
        addDialog.setPositiveButton("Ok"){
            dialog,_->
            val userNames = userName.text.toString()
            val userNumbers = userNo.text.toString()
           userList.add(UserData("Name: $userNames","Mobile Number. : $userNumbers"))
            userAdapater.notifyDataSetChanged()
            Toast.makeText(this,"Adding User Information Sucessfully",Toast.LENGTH_SHORT).show()
            dialog.dismiss()
        }
        addDialog.setNegativeButton("Cancel"){
            dialog,_->
            dialog.dismiss()
            Toast.makeText(this,"Cancel",Toast.LENGTH_SHORT).show()
        }
        addDialog.create()
        addDialog.show()
    }
}